src/components/_classes/component/Component.js

```diff
@@ -3388,14 +3387,15 @@ export default class Component extends Element {
     if (flags.silentCheck) {
       return [];
     }
+    let dirty = this.dirty || flags.dirty
     if (this.options.alwaysDirty) {
-      flags.dirty = true;
+      dirty = true;
     }
-    if (flags.fromSubmission && this.hasValue(data)) {
-      flags.dirty = this.pristine && this.component.protected ? false : true;
+    if (flags.fromSubmission && !_.isUndefined(_.get(flags?.submission?.data, this.key)) && !(this.pristine && this.protected)) {
```

Comment:
_.get(submission.data, this.key) not always the way to fetch the correct data for a component. Particularly the Checkbox components configured as Radio components would fail here. Why can we not use "this.hasValue"?  It looks like you can pass data to it... maybe change this to ```this.hasValue(flags?.submission.data)```?

Reply:
hasValue(data) {
    return !_.isUndefined(_.get(data || this.data, this.key));
  }
hasValue will fallback on this.data. I do not want it to fall back on this.data because I want the explicit submission to set if the dirty flag is true. this.data will be populated with this.empty values in the init function of setting up the form. I need absolutely sure that the property was set by whoever defined the submission object in setSubmission(...). This is why I added the submission to the flags object when a submission is set.